armory/datasets/config_load.py

```diff
@@ -0,0 +1,85 @@
+"""
+Temporary file for testing loading from config without modifying armory.utils
+"""
+
+import copy
+
+from armory.datasets import load, preprocessing, generator
+
+
+def actual_loader(
+    name,
+    version=None,
+    split="test",
+    shuffle_files=False,
+    framework="numpy",
+    epochs=1,
+    drop_remainder=False,
+    num_batches=None,
+    batch_size=1,
+    shuffle_elements=False,
+):
+    info, ds_dict = load.load(name, version=version, shuffle_files=shuffle_files)
+    preprocessor = preprocessing.get(name, version=version)
+    return generator.ArmoryDataGenerator(
+        info,
+        ds_dict,
+        split=split,
+        batch_size=batch_size,
+        framework=framework,
+        epochs=epochs,
+        element_filter=None,
+        element_map=preprocessor,
+        shuffle_elements=shuffle_elements,
+    )
+
+
+def load_dataset(dataset_config, *args, check_run=False, **kwargs):
+    """
+    Designed to be a drop-in replacement for armory.utils.config_loading.load_dataset
+
+    NOTE: very ugly
+    """
+    dataset_config = copy.deepcopy(dataset_config)
+    module = dataset_config.pop("module")
+    if module == "armory.data.datasets":
+        pass
+    elif module == "armory.data.adversarial_datasets":
+        pass  # TODO: check
+    else:
+        raise NotImplementedError
+
+    name = dataset_config.pop("name")
+    if ":" in name:
+        name, version = name.split(":")
+    else:
+        version = None
+
+    for k in "eval_split", "train_split":
```

Comment:
I'd need to look more closely at exactly how we're handling `"eval_split"` and `"train_split"` but this for loop seems a bit messy. I understand, though, that these keys need to be removed before `kwargs` are passed to `actual_loader()`

Reply:
Yes, exactly. The configs can look like this:
```
    "dataset": {
        "batch_size": 1,
        "eval_split": "test_clean",
        "framework": "numpy",
        "module": "armory.data.datasets",
        "name": "librispeech",
        "train_split": "train_clean100"
    },
```

And scenarios map those into the split needed at different points in time.

In contrast, I think we'll probably want to update the config structure so that the training dataset is a completely separate block, which should simplify some of that logic.